onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -258,9 +262,9 @@ void ModelBuilder::RegisterInitializers() {
 }
 
 void ModelBuilder::RegisterModelInputs() {
-  for (int32_t input_idx = 0; input_idx < model_proto_.graph().input_size(); input_idx++) {
-    const auto& input(model_proto_.graph().input(input_idx));
-    std::string input_name = input.name();
+  for (const auto* node_arg : graph_view_.GetInputs()) {
+    ORT_ENFORCE(node_arg != nullptr, "input cannot be null");
+    const auto& input_name = node_arg->Name();
```

Comment:
nit: ORT_ENFORCE throws which adds a little to the binary size. You don't need this check as Graph ensures there are no nullptr instances in this.